src/main/java/qna/service/QnAService.java

```diff
@@ -35,13 +35,9 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        question.checkOwner(loginUser);
 
-        Answers answers = question.getAnswers();
-        answers.checkOwner(loginUser);
+        List<DeleteHistory> deleteHistories = new ArrayList<>(question.delete(loginUser));
```

Comment:
```suggestion
        List<DeleteHistory> deleteHistories = question.delete(loginUser);
```
전달해주는 메소드에서 방어적 복사를 구현하는것을 권장드려요 ! 
불필요한 중복 코드가 발생하지 않을것입니다 😄

Reply:
안녕하세요!
new ArrayList<>() 에서 바로 초기화 하는것보다 
적어주신것처럼 return에서 바로 처리하면 말씀하신것처럼 불필요한 중복 코드는 발생하지 않을거 같은데요.

방어적 복사 관련해서는 두개가 어떤 차이가 있을지 궁금합니다.
ArrayList생성자를 살펴보니 Arrays.copyOf를 사용하고 있는거 같아서 기존 list와는 별개인 새로운 list를 생성하는것처럼 보여지거든요 방어적 복사 관련해서 effective java 내용을 봤는데도 위에 두 코드간 어떤 차이가 있는지 정확하게 이해를 못하겠습니다

Reply:
@sah3122 안녕하세요!
new ArrayList<>() 에서 바로 초기화 하는것보다
적어주신것처럼 return에서 바로 처리하면 말씀하신것처럼 불필요한 중복 코드는 발생하지 않을거 같은데요.

방어적 복사 관련해서는 두개가 어떤 차이가 있을지 궁금합니다.
ArrayList생성자를 살펴보니 Arrays.copyOf를 사용하고 있는거 같아서 기존 list와는 별개인 새로운 list를 생성하는것처럼 보여지거든요 방어적 복사 관련해서 effective java 내용을 봤는데도 위에 두 코드간 어떤 차이가 있는지 정확하게 이해를 못하겠습니다